src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -59,7 +170,7 @@ std::vector<TRShape> shape_infer(const Squeeze* op,
                         return dim.compatible(1);
                     });
                 if (has_squeezable_dim) {
-                    output_shape = PartialShape::dynamic(arg_rank.get_length() - 1);
+                    output_shape = PartialShape::dynamic({arg_rank.get_length() - 1, arg_rank.get_length()});
```

Comment:
Could you explain what is the expected result for such PartialShape definition and point specific unit test for this change?
It looks like Rank object initialized with range Dimension (while rank can be only static or dynamic, ranges for Rank are not supported). Could 

`output_shape = PartialShape::dynamic({arg_rank.get_length() - 1, arg_rank.get_length()})`

Reply:
There are a few test for that particular case: https://github.com/openvinotoolkit/openvino/pull/26995/commits/6450f728ef73d3c12cad47cd6d94ee4f0b44e5f3#diff-17ddb7d4bd361c05e41dd9a2e22a538ddf79520a78bdd0fef7373540fe31ffddR100
If needed I can change it to fully dynamic result instead of expected Rank range.

Reply:
The defined range for the rank has no effect here. 
Such instantiation always leads to fully dynamic dimension, the provided range for the Rank is ignored.
All of the tests with replaced `PartialShape::dynamic({2, 3})` to `PartialShape::dynamic()` will pass.

https://github.com/openvinotoolkit/openvino/blob/961b891ea423427e74b47d24629b33efd866c793/src/core/src/partial_shape.cpp#L174-L176